### PR TITLE
Add fix for one issue with TestFleetRecreateGameServers test

### PR DESCRIPTION
**Pull request by chiayi**
Add fix for one issue with TestFleetRecreateGameServers test

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking

/kind bug

> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:
The test `TestFleetRecreateGameServers` fails due to an issue with the number of gameservers it receives, instead of having 10, it gets 11 gameservers. And one of the gameservers would be in `Shutdown` state. 

**Which issue(s) this PR fixes**:
TestFleetRecreateGameServers flakiness

**Special notes for your reviewer**:

### CODE REVIEW

**Review by gongmax — APPROVED** [review]
Nice catch for this corner case!

File: test/e2e/fleet_test.go [hunk 1]
```diff
@@ -1350,13 +1350,19 @@ func TestFleetRecreateGameServers(t *testing.T) {
 
 			list, err := listGameServers(ctx, flt, client)
 			assert.NoError(t, err)
-			assert.Len(t, list.Items, int(flt.Spec.Replicas))
+			var gameservers []agonesv1.GameServer
+			for _, gs := range list.Items {
+				if gs.Status.State != "Shutdown" {
```
Comment (zmerlynn) [inline test/e2e/fleet_test.go:1355]: nit: use `agonesv1.GameServerStateShutdown`

**Review by zmerlynn — APPROVED** [review]

File: test/e2e/fleet_test.go (hunk 1 above)
Comment (chiayi) [reply test/e2e/fleet_test.go]: Done!

**Review by zmerlynn — APPROVED** [review]
